daphne_worker/src/dap.rs

```diff
@@ -595,6 +595,19 @@ where
     }
 }
 
+fn task_id_from_report(report: &[u8]) -> std::result::Result<Id, DapError> {
+    // The task id is the first 32 bytes of the serialized report.
+    if report.len() < 32 {
+        // Should never happen!
+        return Err(DapError::Fatal(
+            "serialized report is too short".to_string(),
+        ));
+    }
+    let mut b: [u8; 32] = [0; 32];
+    b[..32].copy_from_slice(&report[..32]);
```

Comment:
Slightly simpler (I'm pretty sure this is correct, you'll have to see if cargo disagrees!)
```suggestion
    let id = Id(&report[..33].try_into().map_err(|_| DapErr::fatal("serialize report is too short"))?);
```